### PR TITLE
Create apply_tags

### PR DESCRIPTION
**Pull request by Nosferican**
Draft!

### CODE REVIEW

**Comment by Nosferican** [comment]
```
jovyan@381b43ea26a5:~$ python --version 2>&1 | awk '{print $2}'
3.7.6
jovyan@381b43ea26a5:~$ R --version |  sed -n 1p | awk '{print $3}'
3.6.1
jovyan@381b43ea26a5:~$ julia --version | awk '{print $3}'
1.3.1
jovyan@381b43ea26a5:~$ jupyter-notebook --version | tr -d '\r'
6.0.3
jovyan@381b43ea26a5:~$ jupyterhub --version | tr -d '\r'
1.1.0
jovyan@381b43ea26a5:~$ jupyter-lab --version | tr -d '\r'
1.2.5
```

File: datascience-notebook/hooks/apply_tags [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# Tag the latest build with the short git sha as well as version of key runtimes
+# and packages. 
+GIT_SHA_TAG=${SOURCE_COMMIT:0:12}
+docker tag $IMAGE_NAME "$DOCKER_REPO:$GIT_SHA_TAG"
+PY_VERSION_TAG="python-$(docker run --rm ${IMAGE_NAME} python --version 2>&1 | awk '{print $2}')"
+docker tag $IMAGE_NAME "$DOCKER_REPO:$PY_VERSION_TAG"
+R_VERSION_TAG=="hub-$(docker run --rm -a STDOUT ${IMAGE_NAME} R --version | sed -n 1p | awk '{print $3}')"
+docker tag $IMAGE_NAME "$DOCKER_REPO:${R_VERSION_TAG%%\r}"
+JULIA_VERSION_TAG=="hub-$(docker run --rm -a STDOUT ${IMAGE_NAME} julia --version | awk '{print $3}')"
```
Comment (parente) [inline datascience-notebook/hooks/apply_tags]: I think this is a copy paste error.

```suggestion
JULIA_VERSION_TAG=="julia-$(docker run --rm -a STDOUT ${IMAGE_NAME} julia --version | awk '{print $3}')"
```

File: datascience-notebook/hooks/apply_tags [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -e
+
+# Tag the latest build with the short git sha as well as version of key runtimes
+# and packages. 
+GIT_SHA_TAG=${SOURCE_COMMIT:0:12}
+docker tag $IMAGE_NAME "$DOCKER_REPO:$GIT_SHA_TAG"
+PY_VERSION_TAG="python-$(docker run --rm ${IMAGE_NAME} python --version 2>&1 | awk '{print $2}')"
+docker tag $IMAGE_NAME "$DOCKER_REPO:$PY_VERSION_TAG"
+R_VERSION_TAG=="hub-$(docker run --rm -a STDOUT ${IMAGE_NAME} R --version | sed -n 1p | awk '{print $3}')"
```
Comment (parente) [inline datascience-notebook/hooks/apply_tags]: I think this is a copy paste error.

```suggestion
R_VERSION_TAG=="r-$(docker run --rm -a STDOUT ${IMAGE_NAME} R --version | sed -n 1p | awk '{print $3}')"
```

**Comment by parente** [comment]
@Nosferican Thanks for sending this PR and working through improving it. 🍰 

Let's merge it and see how it goes. If there are problems when Docker Hub attempts to build the image, we might need to iterate. I'll comment back here with the results since the build logs are, unfortunately, private.

**Comment by parente** [comment]
I fixed a small error missed during code review with a direct comment to master: https://github.com/jupyter/docker-stacks/commit/9f4983c5d1f3998a2b3b24583cd03dad4aa5f749#diff-57bd71ca260fa0afd2753f51e94b8420

**Pull request merged**